### PR TITLE
SPORTSHUB-220 fix-authentication-flow-to-work-across-two-devices

### PR DESCRIPTION
**Pull request by inzaenity**
**PLEASE READ**

**Problem:**
Syrio Signed In on one device and logged in on another

**Cause**
I don't think the cause of this is to do with the fact that He used two devices, the sign in and logging in should run independently of another.
I think the cause is due to the lack of atomicity in the logging in Function. 
The Logging in Code runs like this SignIn -> check if user verified -> **if not** send email again -> **if so** check if userData exists -> if user data exists then this is just normal login Proceed -> if userdata does not exists run the create user data workflow from temp data etc
So technically, the user could finish the first part of the workflow and log in, but the code stops running after hence the user is logged in but the userdata isn't populated. This could technically happen if the user closes the browser in the small time frame after the login but before the user Creation (I think this is what happened). 

So the simple solution in theory would be to just sign if after all the logging in code and make sure the logging in code is atomic

**Problem**
Problem is the logging in code requires the firebase object so we need to sign in first then run the logging in code.
2nd Problem Firebase support transactions(atomic database operations) but not cross system operations (etc firebase auth and firebase firestore), there is no easy way to make sure that a workflow containing both auth and firestore are atomic (we could with a backend but...Also can look into 3rd party things such as redis)

**Temporary Makeshift solution**
What I have done is basically if it fails at any point the code, it will automatically log you out and rollback the changes so that hopefully you can try again and succeed the next time

I have also added a lot of logging

### CODE REVIEW

File: frontend/services/src/auth/authService.ts [hunk 1]
```diff
@@ -27,12 +27,12 @@ export async function handleEmailAndPasswordSignUp(data: NewUserData) {
...
 
-    // Send email verification
-    await sendEmailVerification(userCredential.user, actionCodeSettings);
-    console.log("Email verification sent. Please verify your email before logging in.");
     const { password, ...userDataWithoutPassword } = data;
     // Save user data temporarily in your database
     await saveTempUserData(userCredential.user.uid, userDataWithoutPassword);
+    // Send email verification
+    await sendEmailVerification(userCredential.user, actionCodeSettings);
+    console.log("Email verification sent. Please verify your email before logging in.");
```
Comment (owya490) [inline frontend/services/src/auth/authService.ts:35]: convert to logger

File: frontend/services/src/auth/authService.ts [hunk 2]
```diff
@@ -49,44 +49,103 @@ export async function handleSignOut(setUser: (user: UserData) => void) {
...
-    // Check if the user's email is verified
-    if (userCredential.user.emailVerified) {
-      console.log("Email is verified. Logging in...");
+    // Sign in with email and passwor
+
+    userCredential = await signInWithEmailAndPassword(auth, email, password);
+    authServiceLogger.info("User Object gotten in sign in workflow", { email, userId: userCredential.user.uid });
+
+    if (!userCredential.user.emailVerified) {
+      authServiceLogger.warn("Email is not verified. Sending verification email.", { userId: userCredential.user.uid });
```
Comment (owya490) [inline frontend/services/src/auth/authService.ts:62]: this is not a warn i believe. Email verification is an expected pathway

File: frontend/services/src/auth/authService.ts [hunk 3]
```diff
@@ -49,44 +49,103 @@ export async function handleSignOut(setUser: (user: UserData) => void) {
...
 
       try {
-        const userExist = await getPublicUserById(userCredential.user.uid);
-        return true;
+        await getPublicUserById(userCredential.user.uid);
+        return true; // User exists, sign-in successful
       } catch (error: unknown) {
-        // Retrieve the temporary user data
         if (error instanceof UserNotFoundError) {
+          authServiceLogger.warn("User not found in public users. Attempting to retrieve temporary user data.", {
```
Comment (owya490) [inline frontend/services/src/auth/authService.ts:73]: this should not be a warn as well as this is expected path

File: frontend/services/src/auth/authService.ts [hunk 4]
```diff
@@ -49,44 +59,103 @@ export async function handleSignOut(setUser: (user: UserData) => void) {
...
+export async function handleEmailAndPasswordSignIn(email: string, password: string): Promise<boolean> {
+  let userCredential: UserCredential | undefined = undefined;
+
   try {
-    const userCredential: UserCredential = await signInWithEmailAndPassword(auth, email, password);
-    console.log("isverified", userCredential.user.emailVerified);
-    // Check if the user's email is verified
-    if (userCredential.user.emailVerified) {
-      console.log("Email is verified. Logging in...");
+    // Sign in with email and passwor
```
Comment (owya490) [inline frontend/services/src/auth/authService.ts:66]: /nit - password

**Review by owya490 — APPROVED** [review]
looks good from a code point.

how have you tested this?

**Review by brian2w — APPROVED** [review]

**Pull request merged**